src/lib/utils.js

```diff
@@ -374,3 +377,13 @@ export const addMissingFields = (submitted, original) => {
 		}
 	}
 };
+
+export const buddiesAreValid = (submitted) => {
+	let userIds = Object.keys(get(users));
+	for (let id of submitted.buddies) {
+		if (!userIds.includes(id)) {
+			return false;
+		}
+	}
```

Comment:
how does this happen?

Reply:
If the user enters an unknown user name in the buddy field

Reply:
In the user enters an unknown buddy name in the buddy field and hits submit, then the id will actually be undefined here, so technically we could just check that all ids are non-null.  But checking that the id actually matches a known user seemed a bit more robust and without costing much.